R/RLum.Analysis-class.R

```diff
@@ -912,40 +915,78 @@ setMethod(
         message("[sort_RLum()]: Only the first field will be used in sorting")
         slot <- slot[1]
       }
-      sort.by.slot <- TRUE
     }
+
     if (!is.null(info_element)) {
       .validate_class(info_element, "character", extra = "NULL")
-      valid.names <- names(object@records[[1]]@info)
+      valid.names <- c("XY_LENGHT", "X_MIN", "X_MAX", "Y_MIN", "Y_MAX", names(object@records[[1]]@info))
       if (any(!info_element %in% valid.names)) {
         .throw_error("Invalid 'info_element' name, valid names are: ",
                      .collapse(valid.names))
       }
-      if (length(info_element) > 1) {
-        message("[sort_RLum()]: Only the first field will be used in sorting")
-        info_element <- info_element[1]
-      }
     }
-    if (is.null(slot) && is.null(info_element)) {
+
+    if (is.null(slot) && all(is.null(info_element))) {
```

Comment:
Given that `info_element` is a character vector, it doesn't seem to make sense to use `all()`, as it's either `NULL` or it contains some element (a vector cannot contain `NULL` elements alongside other valid ones).